src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -85,4 +131,338 @@ public void requestMessageFromIris(IrisSession session) {
         // TODO: Future: Switch between different session types
         irisChatService.requestAndHandleResponse(session);
     }
+
+    /**
+     * Creates the initial system message for the session and set LLM
+     *
+     * @param session The session to generate the initial system message for
+     * @return The created IrisMessage for storing in the database
+     */
+    public IrisMessage createInitialSystemMessage(IrisSession session) {
+        var exercise = session.getExercise();
+        String title = "undefined";
+        if (exercise.getCourseViaExerciseGroupOrCourseMember() != null && exercise.getCourseViaExerciseGroupOrCourseMember().getTitle() != null) {
```

Comment:
In the future we want proper error handling for this, so maybe adding a TODO here would be good.